spec/request/address_request_spec.rb

```diff
@@ -9,153 +9,173 @@
-      it 'should match the errors schema' do
-        VCR.use_cassette('evss/pciu_address/address_500') do
-          get '/v0/address', nil, auth_header
-          expect(response).to have_http_status(:bad_gateway)
-          expect(response).to match_response_schema('errors')
+      context 'with a 500 response' do
+        it 'should match the errors schema' do
+          VCR.use_cassette('evss/pciu_address/address_500') do
+            get '/v0/address', nil, auth_header
+            expect(response).to have_http_status(:bad_gateway)
+            expect(response).to match_response_schema('errors')
+          end
         end
       end
     end
-  end
 
-  describe 'GET /v0/address/rds/states' do
-    context 'with a 200 response' do
-      it 'should match the states schema' do
-        VCR.use_cassette('evss/reference_data/states') do
-          get '/v0/address/rds/states', nil, auth_header
-          expect(response).to have_http_status(:ok)
-          expect(response).to match_response_schema('states')
+    describe 'GET /v0/address/states' do
+      context 'with a 200 response' do
+        it 'should match the states schema' do
+          VCR.use_cassette('evss/pciu_address/states') do
+            get '/v0/address/states', nil, auth_header
+            expect(response).to have_http_status(:ok)
+            expect(response).to match_response_schema('states')
+          end
         end
       end
     end
-  end
 
-  describe 'GET /v0/address/rds/countries' do
-    context 'with a 200 response' do
-      it 'should match the states schema' do
-        VCR.use_cassette('evss/reference_data/countries') do
-          get '/v0/address/rds/countries', nil, auth_header
-          expect(response).to have_http_status(:ok)
-          expect(response).to match_response_schema('countries')
+    describe 'GET /v0/address/countries' do
+      context 'with a 200 response' do
+        it 'should match the countries schema' do
+          VCR.use_cassette('evss/pciu_address/countries') do
+            get '/v0/address/countries', nil, auth_header
+            expect(response).to have_http_status(:ok)
+            expect(response).to match_response_schema('countries')
+          end
         end
       end
     end
   end
 
-  describe 'GET /v0/address/states' do
-    context 'with a 200 response' do
-      it 'should match the states schema' do
-        VCR.use_cassette('evss/pciu_address/states') do
-          get '/v0/address/states', nil, auth_header
-          expect(response).to have_http_status(:ok)
-          expect(response).to match_response_schema('states')
+  context '#reference_data_service.enabled=true' do
+    before do
+      Settings.evss.reference_data_service.enabled = true
+      Rails.application.reload_routes!
+    end
+
+    describe 'GET /v0/address/countries' do
+      context 'with a 200 response' do
+        it 'should match the states schema' do
+          VCR.use_cassette('evss/reference_data/countries') do
```

Comment:
Is there any way we can set up the same request spec for both conditions? It would be great if the request spec could validate that the response schema was the same regardless of the `reference_data_service.enabled` flag.

Reply:
> It would be great if the request spec could validate that the response schema was the same regardless of the reference_data_service.enabled flag

Not entirely sure I follow - I _think_ what you're asking for is already being done.  The diff in this file is mostly from tabulation which makes it tricky to read. Running `rspec ./spec/request/address_request_spec.rb -fdoc` yields:
```
# edited down to just the important parts...

address
  #reference_data_service.enabled=false
    GET /v0/address/states
      with a 200 response                                                                                                                                                                                                                                                                                                                                                  
        should match the states schema                                                                                                                                                                                                                                                                                                                                     
    GET /v0/address/countries                                                                                                                                                                                                                                                                                                                                              
      with a 200 response                                                                                                                                                                                                                                                                                                                                                  
        should match the countries schema                                                                                                                                                                                                                                                                                                                                  
  #reference_data_service.enabled=true                                                                                                                                                                                                                                                                                                                                     
    GET /v0/address/states                                                                                                                                                                                                                                                                                                                                                 
      with a 200 response                                                                                                                                                                                                                                                                                                                                                  
        should match the states schema                                                                                                                                                                                                                                                                                                                                               
    GET /v0/address/countries                                                                                                                                                                                                                                                                                                                                              
      with a 200 response                                                                                                                                                                                                                                                                                                                                                  
        should match the states schema
```

Our two endpoints (using all 4 evss endpoints), are checked against the same 2 schemas.